core/src/main/java/hudson/Functions.java

```diff
@@ -2021,6 +2061,10 @@ public String getPasswordValue(Object o) {
                 if (item != null && !item.hasPermission(Item.CONFIGURE)) {
                     return "********";
                 }
+                Computer computer = req.findAncestorObject(Computer.class);
+                if (computer != null && !computer.hasPermission(Computer.CONFIGURE)) {
+                    return "********";
```

Comment:
❤️

Reply:
I do not know a plugin which would be wrongly affected by that.
I can imagine a Computer action which has a form with credentials, but it is much more likely for Jobs which has been added 4 years ago